### ISSUE TITLE
The accessibility test falsely fails because the shadowDom cannot be accessed

### ISSUE THREAD

**Issue by mugishajk** [dequelabs/axe-core#4121]
### Product

axe-core

### Product Version

4.6.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
The test is expected to pass since the child `li` is directly in the parent `ol`

#### Actual
The test fails stating that the structure is not accessible since `li` should be a direct child of `ol` 

#### How to Reproduce

- make a parent component with `<ol><slot></slot></ol> `as a base tag with a default slot as direct child
- make a child component  with `<li>my child component</li>` as base tag
- run the accessibility tests on the composed component `<parent><child>one</child><child>two</child></parent>`
`await expect(el).to.be.accessible();`
